code/_onclick/xeno.dm

```diff
@@ -37,12 +37,34 @@
 		if(XENO_NO_DELAY_ACTION)
 			next_move = world.time
 		else
-			if(!tile_attack)
+			if(!tile_attack) // Patting flames for Xenos
+				var/firepatted = FALSE
+				if(src.a_intent == INTENT_HELP)
+					var/fire_level_to_extinguish = 5
+					var/turf/T = target
+					for(var/obj/flamer_fire/FF in T)
+						firepatted = TRUE
+						if((FF.firelevel > fire_level_to_extinguish) && (!FF.fire_variant)) //If fire_variant = 0, default fire extinguish behavior.
+							FF.firelevel -= fire_level_to_extinguish
+							FF.update_flame()
+						else
+							switch(FF.fire_variant)
+								if(FIRE_VARIANT_TYPE_B) //Armor Shredding Greenfire, extinguishes faster.
+									if(FF.firelevel > 2*fire_level_to_extinguish)
+										firepatted = TRUE
+										FF.firelevel -= 2*fire_level_to_extinguish
+										FF.update_flame()
+									else qdel(FF)
+								else qdel(FF)
 				xeno_miss_delay(src)
 				animation_attack_on(target)
 				playsound(loc, 'sound/weapons/alien_claw_swipe.ogg', 10, 1) //Quiet to limit spam/nuisance.
-				visible_message(SPAN_DANGER("[src] swipes at [target]!"), \
-				SPAN_DANGER("You swipe at [target]!"), null, 5, CHAT_TYPE_XENO_COMBAT)
+				if(firepatted)
+					visible_message(SPAN_DANGER("[src] pats at the fire!"), \
```

Comment:
these don't show up when patting fire, pretty sure you're missing this below but i may be wrong
```suggestion
					src.visible_message(SPAN_DANGER("[src] pats at the fire!"), \
```